meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -30,19 +44,65 @@ interface IShowStyleVariantsProps {
 interface IShowStyleVariantsSettingsState {
 	editedMappings: ProtectedString<any>[]
 	timestampedFileKey: number
+	dndVariants: ShowStyleVariant[]
 }
 
+interface DragVariant {
+	index: number
+	type: ShowStyleDragDropTypes
+}
+
+const SET_STATE_DELAY: number = 50
+
 export const ShowStyleVariantsSettings = withTranslation()(
 	class ShowStyleVariantsSettings extends React.Component<
 		Translated<IShowStyleVariantsProps>,
 		IShowStyleVariantsSettingsState
 	> {
+		private timer: number | undefined
+
 		constructor(props: Translated<IShowStyleVariantsProps>) {
 			super(props)
 
 			this.state = {
 				editedMappings: [],
 				timestampedFileKey: Date.now(),
+				dndVariants: this.props.showStyleVariants,
+			}
+		}
+		componentDidUpdate(prevProps: Readonly<Translated<IShowStyleVariantsProps>>) {
+			if (this.showStyleVariantsChanged(prevProps) || this.noShowStyleVariantsPresentInState()) {
+				this.timer = Meteor.setTimeout(() => {
+					this.setState({
+						dndVariants: this.props.showStyleVariants,
+					})
+				}, SET_STATE_DELAY)
+			} else if (this.timer) {
+				Meteor.clearTimeout(this.timer)
+			}
+		}
+
+		componentWillUnmount() {
+			if (this.timer) {
+				Meteor.clearTimeout(this.timer)
+			}
+		}
+
+		private showStyleVariantsChanged = (prevProps: Readonly<Translated<IShowStyleVariantsProps>>): boolean => {
+			if (prevProps.showStyleVariants.length !== this.props.showStyleVariants.length) {
```

Comment:
Swapping the order of the properties being checked, should improve the methods readability slightly, as the same property will be above one another.
i.e change the `if` too: `if (this.props.showStyleVariants.length !== prevProps.showStyleVariants.length)`.